src/lexers/jsx-lexer.js

```diff
@@ -54,4 +58,59 @@ export default class JsxLexer extends HTMLLexer {
 
     return this.keys
   }
+
+  /**
+   * Recursively convert html tags and js injections to tags with the child index in it
+   *
+   * @param {string} string
+   *
+   * @returns string
+   */
+  eraseTags(string) {
+    const children = this.simplify(acorn.parse(string, {plugins: {jsx: true}}).body[0].expression.children, string);
+
+    const elemsToString = children => children.map((child, index) => {
+      switch(child.type) {
+        case 'text': return child.content;
+        case 'js': return `<${index}>${child.content}</${index}>`;
+        case 'tag': return `<${index}>${elemsToString(child.children)}</${index}>`;
+        default: throw new ParsingError('Unknown parsed content: ' + child.type);
+      }
+    }).join('');
+
+    return elemsToString(children);
+  }
+
+  /**
+   * Simplify the bulky AST given by Acorn
+   * @param {*} children An array of elements contained inside an html tag
+   * @param {string} originalString The original string being parsed
+   */
+  simplify(children, originalString) {
+    for (let i = 0; i < children.length; i ++) {
```

Comment:
Is there a reason you don't use `forEach` here? If not, it would be more consistent with the rest of the code